### DISCUSSION TITLE
eza-ls wrapper by @kesor

### DISCUSSION THREAD

**Discussion by glensc** [eza-community/eza#1283] [Q&A, Howto, Help] [answered]
@kesor has published nice wrapper:
- https://github.com/eza-community/eza/issues/736#issuecomment-2556709185

Re-post it as a discussion so as not to hijack #736 too much.

**Comment by glensc** [comment]
@kesor: I've noticed that I use `ls -l -rtc` to get recent files by creation time last, and your wrapper doesn't get it right (the same with ls)

to me looks like the `-r` flag is reversed, if I omit it, it seems to result expected output

**Reply by kesor** [reply]
Right, I changed it from `"--sort=modified"` to `"--sort=old"`; Because that is the default sorting for `ls`. Which makes reverse work as expected. I can't edit the original comment anymore, since that discussion was closed.

There are other sorting options in `ls` that are reversed in the script. For example, `ls -l -S` would sort by size from large files to small files. With the wrapper, eza is doing the opposite.

**Reply by glensc** [reply]
maybe publish the script as gist? or just new repo?

**Reply by cafkafk** [reply]
we could make a repo in the @eza-community org for that script, anyone interested in that? (ideally someone that isn't me would maintain it)

**Reply by kesor** [reply, marked as answer]
https://gist.github.com/kesor/6e094187c85f05c30fc923d6923c1073

**Reply by glensc** [reply]
@kesor can you address `ls -lrtc`?

I'm expecting to see files by last created, reverse order  and display created time

**Reply by kesor** [reply]
@glensc I modified the gist, now `-c` is still `--changed`, but when used with `-tc` it is now `--created`.

**Comment by glensc** [comment]
# support: `-L` flag

in `ls` the `-L` flag dereferences symlinks

eza behavior:
1. `ls -l -L translations.json` shows full dir listing
2. `ls -l translations.json -L`, gives error: `eza: Flag -L needs a value`

**Reply by kesor** [reply]
Updated the gist. If you do want to use `-L` from eza, use `--level` instead.

**Reply by glensc** [reply]
duh, `eza` offers no `ls -L` functionality?  I tried `--no-symlinks`, `--follow-symlinks`, but neither of them do the needed thing.

**Reply by kesor** [reply]
I'm using version v0.20.11 and it has these -

```
--show-symlinks            explicitly show symbolic links (for use with --only-dirs | --only-files)
--no-symlinks              do not show symbolic links
-L, --level DEPTH          limit the depth of recursion
```

It shows the symlinks for me in `ls -l` like this,

```
lrwxrwxrwx    - evgeny  4 Jul  2020  bin -> /home/evgeny/src/dotfiles/bin/
```

**Reply by glensc** [reply]
you seem have misunderstood, `ls -L` does the opposite, it will show the file symlink points to like it's a file there.

```sh
$ ln -s /etc/passwd /tmp/passwd.example
$ ls  -l /tmp/passwd.example
lrwxr-xr-x 1 glen wheel 11 Jan 15 23:31 /tmp/passwd.example -> /etc/passwd

$ ls -l -L /tmp/passwd.example /etc/passwd
-rw-r--r-- 1 root wheel 6946 Oct 18  2019 /etc/passwd
-rw-r--r-- 1 root wheel 6946 Oct 18  2019 /tmp/passwd.example
```

from `man ls`:

> 
>       -L, --dereference
>              when  showing file information for a symbolic link, show information for the file the link references rather than for the
>              link itself

**Reply by kesor** [reply]
I changed the gist so that it replaces `-L` with `-X`.

**Reply by glensc** [reply]
thanks. minor nitpick: ` # Suppress -L` comment is now outdated.

**Reply by kesor** [reply]
No, it still swallows the `-L` and instead the `-L` adds `-X` to arguments. So the comment is right in place, just like the other ones which mention suppressing other options.
